--- logilab-astng-0.21.1.orig/scoped_nodes.py   2011-01-11 15:47:29.000000000 +0100
+++ logilab-astng-0.21.1/scoped_nodes.py        2011-07-08 11:07:34.000000000 +0200
@@ -324,6 +324,8 @@
 
     def import_module(self, modname, relative_only=False, level=None):
         """import the given module considering self as context"""
+        if relative_only and level is None:
+            level = 0
         absmodname = self.relative_to_absolute_name(modname, level)
         try:
             return MANAGER.astng_from_module_name(absmodname)
@@ -342,6 +344,8 @@
         # XXX this returns non sens when called on an absolute import
         # like 'pylint.checkers.logilab.astng.utils'
         # XXX doesn't return absolute name if self.name isn't absolute name
+        if self.absolute_import_activated() and level is None:
+            return modname
         if level:
             if self.package:
                 level = level - 1
