--- logilab-astng-0.19.3/inference.py.orig	2009-12-18 11:38:51.000000000 +0100
+++ logilab-astng-0.19.3/inference.py	2010-03-23 09:23:13.000000000 +0100
@@ -50,9 +50,7 @@
 _CONST_PROXY[type(None)].parent = _CONST_PROXY[bool].parent
 
 def _set_proxied(const):
-    if not hasattr(const, '__proxied'):
-        const.__proxied = _CONST_PROXY[const.value.__class__]
-    return const.__proxied
+    return _CONST_PROXY[const.value.__class__]
 nodes.Const._proxied = property(_set_proxied)
 
 nodes.Const.__bases__ += (Instance,)
